Feature/FIT-7-Login

Ahmed-Abdelghany-11 opened the pull request:
This pull request implements the **User Login** functionality, integrating it seamlessly across the **data**, **domain**, and **presentation** layers. This enhancement enables existing users to securely log into their accounts with proper form validation, local and remote data handling, and backend communication. It also strengthens architecture consistency and improves overall code quality.

### Changes Included:

#### 🔹 Data Layer:
- Added login API endpoint in `AuthRetrofitClient`.
- Implemented **Auth Remote Data Source** to manage login API calls.
- Implemented **Auth Local Data Source** for saving login data locally.
- Created request and response DTOs for:
  - `LoginRequestDto`
  - `LoginResponseDto`
- Developed the **Auth Repository Implementation** to handle login logic, data persistence, and error handling.

#### 🔹 Domain Layer:
- Added a **use case** for user login.
- Updated **repository contracts** to support login logic.

#### 🔹 Presentation Layer:
- Developed the **Login Screen** with input fields and form validation.
- Integrated **Bloc/Cubit** for managing login state.
- Implemented loading indicators, error display, and navigation logic for better UX.

#### 🔹 Navigation & UI Enhancements:
- Updated navigation to **Register Screen** from the `DonotHaveAccountText` widget.
- Applied consistent styling and validation feedback to input fields.

#### 🔹 General Improvements:
- **Exception Handling:**
  - Integrated error parsing in the login flow.
  - Display meaningful error messages from the backend.

#### 🔹 Localization:
- Updated `ar.json` and `en.json` to include **new login-related strings**.

## 📌 Pull Request Checklist

### ✅ General
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] The PR title follows the required format: `feat/FIT-7-Login`.
- [x] The branch name follows the required format: `feature/fit-7-login`.

### 📂 Code Quality
- [x] All filenames are in `snake_case`.
- [x] All functions and variable names follow Dart naming conventions.
- [x] All new code is formatted with `dart format .`

### 🛠 CI/CD Validation
- [x] Linting passes with `dart analyze`.
- [x] The branch name is validated against the allowed patterns.
- [x] The PR title is validated against the required format.

### 🚀 Testing
- [ ] Unit tests are included for all new features and bug fixes.
- [ ] `flutter test --coverage` has been run and the results reviewed.
- [ ] The project builds successfully without errors.

### 📝 Additional Information
This update delivers the **complete user login flow**, focusing on clean architecture, error resilience, and responsive UI/UX. Future work will enhance test coverage and improve session management.

### 🔗 Related JIRA Ticket(s):
- [x] JIRA Issue: `FIT-7`

### 📸 Screenshots (if applicable):

PeshoOReda committed:
feat(auth): implement login functionality with UI and state management

This commit introduces the login feature, including:

- **Data Layer:**
    - `LoginRequestDto` and `LoginResponseDto` for API communication.
    - `UserDto` for user data representation.
    - Updated `AuthRetrofitClient` with login endpoint.
    - Implemented `AuthRepoImpl` with login logic and token saving.
    - Implemented `AuthLocalDataSourceImpl` for token storage.
    - Implemented `AuthRemoteDataSourceImpl` for API calls.
- **Domain Layer:**
    - `LoginRequest` entity.
    - `UserEntity` for domain representation.
    - `LoginUseCase` for business logic.
- **Presentation Layer (Login Screen):**
    - `LoginScreen` UI with email/password fields, login button, "Forgot Password", and "Don't have an account" options.
    - `LoginCubit` for state management, handling login requests and navigation.
    - `LoginAction` and `LoginState` for Cubit communication.
    - Individual widgets for UI components: `LoginFields`, `ForgotPasswordButton`, `LoginButton`, `DoNotHaveAccount`.
- **Core:**
    - Added `BaseNavigationState` and `NavigationType` to `BaseState` for handling navigation actions.
    - Updated `AppRoutes` with the login route.
    - Renamed `inital_route_function.dart` to `initial_route_function.dart`.
- **DI:**
    - Updated dependency injection configuration for new classes.

Ahmed-Abdelghany-11 committed:
feat(auth): implement local and remote data sources for authentication

Ahmed-Abdelghany-11 committed:
feat(auth): implement login functionality with local and remote data sources

Ahmed-Abdelghany-11 committed:
feat(auth): implement login cubit with form validation and state management

Ahmed-Abdelghany-11 committed:
feat(login): add login screen UI and integrate with view model

Ahmed-Abdelghany-11 committed:
feat(auth): update navigation to register screen in DonotHaveAccount widget

Ahmed-Abdelghany-11 committed:
refactor(auth): clean up code formatting and variable naming in authentication files

PeshoOReda committed:
Feature/Fit-7-Login

PeshoOReda committed:
feat(auth): implement login functionality with UI and state management

This commit introduces the login feature, including:

- **Data Layer:**
    - `LoginRequestDto` and `LoginResponseDto` for API communication.
    - `UserDto` for user data representation.
    - Updated `AuthRetrofitClient` with login endpoint.
    - Implemented `AuthRepoImpl` with login logic and token saving.
    - Implemented `AuthLocalDataSourceImpl` for token storage.
    - Implemented `AuthRemoteDataSourceImpl` for API calls.
- **Domain Layer:**
    - `LoginRequest` entity.
    - `UserEntity` for domain representation.
    - `LoginUseCase` for business logic.
- **Presentation Layer (Login Screen):**
    - `LoginScreen` UI with email/password fields, login button, "Forgot Password", and "Don't have an account" options.
    - `LoginCubit` for state management, handling login requests and navigation.
    - `LoginAction` and `LoginState` for Cubit communication.
    - Individual widgets for UI components: `LoginFields`, `ForgotPasswordButton`, `LoginButton`, `DoNotHaveAccount`.
- **Core:**
    - Added `BaseNavigationState` and `NavigationType` to `BaseState` for handling navigation actions.
    - Updated `AppRoutes` with the login route.
    - Renamed `inital_route_function.dart` to `initial_route_function.dart`.
- **DI:**
    - Updated dependency injection configuration for new classes.

PeshoOReda committed:
Feature/Fit-7-Login

PeshoOReda committed:
Merge remote-tracking branch 'origin/feature/fit-7-login' into feature/fit-7-login

# Conflicts:
#	lib/core/assets/app_images.dart
#	lib/core/utils/di/di.config.dart
#	lib/core/utils/routes/app_routes.dart
#	lib/data/auth/api/auth_retrofit_client.dart
#	lib/data/auth/data_source/contract/auth_local_data_source.dart
#	lib/data/auth/data_source/contract/auth_remote_data_source.dart
#	lib/data/auth/data_source/local/auth_local_data_source_impl.dart
#	lib/data/auth/data_source/remote/auth_remote_data_source_impl.dart
#	lib/data/auth/repo_impl/auth_repo_impl.dart
#	lib/domain/auth/repo/auth_repo.dart
#	lib/domain/auth/use_case/login_use_case.dart
#	lib/features/login/presentation/view/login_screen.dart
#	lib/features/login/presentation/view_model/cubit/login_cubit.dart
#	lib/features/login/presentation/view_model/cubit/login_state.dart

PeshoOReda committed:
feat(auth): implement login functionality with UI and state management

This commit introduces the login feature, including:

- **Data Layer:**
    - `LoginRequestDto` and `LoginResponseDto` for API communication.
    - `UserDto` for user data representation.
    - Updated `AuthRetrofitClient` with login endpoint.
    - Implemented `AuthRepoImpl` with login logic and token saving.
    - Implemented `AuthLocalDataSourceImpl` for token storage.
    - Implemented `AuthRemoteDataSourceImpl` for API calls.
- **Domain Layer:**
    - `LoginRequest` entity.
    - `UserEntity` for domain representation.
    - `LoginUseCase` for business logic.
- **Presentation Layer (Login Screen):**
    - `LoginScreen` UI with email/password fields, login button, "Forgot Password", and "Don't have an account" options.
    - `LoginCubit` for state management, handling login requests and navigation.
    - `LoginAction` and `LoginState` for Cubit communication.
    - Individual widgets for UI components: `LoginFields`, `ForgotPasswordButton`, `LoginButton`, `DoNotHaveAccount`.
- **Core:**
    - Added `BaseNavigationState` and `NavigationType` to `BaseState` for handling navigation actions.
    - Updated `AppRoutes` with the login route.
    - Renamed `inital_route_function.dart` to `initial_route_function.dart`.
- **DI:**
    - Updated dependency injection configuration for new classes.

ali72-20 commented on lib/features/login/presentation/view/widget/login_field.dart:
```diff
@@ -0,0 +1,86 @@
+          validator: (value) =>
+              widget.viewModel.validator.validateEmail(value ?? ""),
+          style: const TextStyle(color: AppColors.white),
+          decoration: InputDecoration(
```
try to setup this decoration in app Theme

ali72-20 reviewed (commented):
good work

ali72-20 reviewed (approved):
good work

PeshoOReda committed:
feat(login): refactor login flow and UI components for improved structure and usability

PeshoOReda committed:
Refactor login screen and assets

- Updated AppIcons to include mail, lock, and eye logos.
- Removed mail, lock, and eye logos from AppImages.
- Added MainLayoutScreen to app routes.
- Refactored LoginScreen to use Image.asset for app logo and replaced LoginBody with LoginForm.
- Cleaned up imports in LoginScreen and other related files.
- Deleted unused LoginBody widget.
- Updated LoginButton to use constant padding and font size.
- Refactored LoginForm to use AppIcons for input field icons.
- Added mail.svg asset for email icon.

PeshoOReda committed:
feat(login): enhance validation with debug messages and improve form structure

ahmedelazab1220 commented on lib/core/utils/routes/app_routes.dart:
```diff
@@ -1,15 +1,30 @@
+      appBar: AppBar(title: const Text('Main Layout')),
+      body: const Center(child: Text('Welcome to the Main Layout!')),
+    );
+  }
```
what do u doing ?

mainLayout screen in appRoutes?!

ahmedelazab1220 committed:
Merge branch 'development' of github.com:ahmedelazab1220/fitness_app into feature/fit-7-login

ahmedelazab1220 committed:
feat(onboarding): enhance onboarding flow and improve navigation; refactor login components

ahmedelazab1220 committed:
feat(auth): reorganize login models and update DTOs; remove deprecated entities

ahmedelazab1220 reviewed (commented):
not follow standard
